overlord/configstate/hooks.go

```diff
@@ -133,3 +108,103 @@ func (h *configureHandler) Done() error {
 func (h *configureHandler) Error(err error) (bool, error) {
 	return false, nil
 }
+
+// defaultConfigureHandler is the handler for the default-configure hook.
+type defaultConfigureHandler struct {
+	context *hookstate.Context
+}
+
+func newDefaultConfigureHandler(context *hookstate.Context) hookstate.Handler {
+	return &defaultConfigureHandler{context: context}
+}
+
+// Before is called by the HookManager before the default-configure hook is run.
+func (h *defaultConfigureHandler) Before() error {
+	h.context.Lock()
+	defer h.context.Unlock()
+
+	tr := ContextTransaction(h.context)
+
+	instanceName := h.context.InstanceName()
+	st := h.context.State()
+	info, err := snapstate.CurrentInfo(st, instanceName)
+	if err != nil {
+		return err
+	}
+	hasDefaultConfigureHook := info.Hooks["default-configure"] != nil
+	hasConfigureHook := info.Hooks["configure"] != nil
+
+	// default-configure hook should only be taken into account if there is also a configure hook,
+	// because it is only intended to provide additional configuration support
+	if hasDefaultConfigureHook && !hasConfigureHook {
+		// returning HookNotUsableError instructs the hook manager to skip a hook and swallow the error
+		return &hookstate.HookNotUsableError{Err: "default-cofigure hook requires configure hook"}
```

Comment:
I know I suggested this behavior but I wonder now if it wouldn't be simpler to error on this in check_snap.go code, and simply fail the installation, thinking a bit more is a bit unclear why somebody purposefully would have create default-configure but leave configure?  We really don't have a precedent with hooks that expect another hook

Reply:
Agreed. 
I left the check in here thinking that it probably good to have in place for coherent behaviour e.g. in testing when mocking sidestep real life steps, but would like to know your preference.